src/Option.ts

```diff
@@ -53,9 +53,30 @@ export class None<A> {
   mapNullable<B>(f: (a: A) => B | null | undefined): Option<B> {
     return none
   }
+
+  /**
+   * `ap`, some may also call it "apply". Takes a function `fab` that is in the context of `Option`, and applies that
+   * function to this `Option`'s value. If the `Option` calling `ap` is `none` it will return `none`.
+   *
+   * For example:
+   *
+   * `const someFn = some(2).ap(some(x => x + 1))` will return `some(3)`.
+   *
+   * `const someFn = none.ap(some(x => x + 1))` will return `none`.
+   */
   ap<B>(fab: Option<(a: A) => B>): Option<B> {
     return none
   }
+  /**
+   * Similar to `ap` but instead of taking a function it takes `some` value or `none`, then applies this `Option`'s
+   * wrapped function to the `some` or `none`. If the `Option` calling `ap_` is `none` it will return `none`.
+   *
+   * For example:
+   *
```

Comment:
You can use `@example` tag, it's recognized as preformatted code by some editors/IDEs